src/OpenTelemetry/Trace/ExceptionProcessor.cs

```diff
@@ -0,0 +1,80 @@
+// <copyright file="ExceptionProcessor.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Diagnostics;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Runtime.InteropServices;
+
+namespace OpenTelemetry.Trace
+{
+    internal class ExceptionProcessor : BaseProcessor<Activity>
+    {
+        private const string ExceptionPointersKey = "otel.exception_pointers";
+
+        private readonly Func<IntPtr> fnGetExceptionPointers;
+
+        public ExceptionProcessor()
+        {
+            try
+            {
+                var flags = BindingFlags.Static | BindingFlags.Public;
+                var method = typeof(Marshal).GetMethod("GetExceptionPointers", flags, null, new Type[] { }, null);
+                var lambda = Expression.Lambda<Func<IntPtr>>(Expression.Call(method));
+                this.fnGetExceptionPointers = lambda.Compile();
+            }
+            catch (Exception ex)
+            {
+                throw new NotSupportedException("System.Runtime.InteropServices.Marshal.GetExceptionPointers is not supported.", ex);
```

Comment:
this may not indicate whats actionable for user. Is turning off `SetErrorStatusOnException` the right mitigation?

Reply:
I think this is implied by the callstack associated with the exception?

Reply:
Trying to explain more about my understanding - I guess:
* `NotSupportedException` is a self-contained message, telling the caller that "such invocation is not supported".
* `NotSupportedException` could provide an optional message, explaining why it is not supported, but not "what's the recommendation".

Reply:
I've updated the wording based on offline discussion with @cijothomas.
I've chosen "Automatic exception detection is not supported on this platform." instead of "This platform does not support SetErrorStatusOnException" since the class which throws the exception has no knowledge of the `SetErrorStatusOnException` flag.